### ISSUE TITLE
[CT-1931] [Bug] Tests fail with DBT Internal Error when specified with meta attribute

### ISSUE THREAD

**Issue by nishantnathhoney** [dbt-labs/dbt-core#6767]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Specifying `meta` attribute for tests leads to `dbt internally failed to execute` error

### Expected Behavior

Gracefully exit the run with warning or error noting `meta` is not expected in tests
or
allow `meta` attribute to be specified for tests as well

### Steps To Reproduce

test_sample.sql
```
{{
  config(
    materialized = 'table',
    meta = {
      'owner': 'randomstring'
    }
  )
}}

SELECT 1 AS column_a
```

`dbt test -s test_sample`

Result: `dbt internally failed to execute test.honey_dbt.test_sample: Returned 0 rows, but expected 1 row`

### Relevant log output

```shell
Internal error executing None
dbt internally failed to execute test.honey_dbt.test_sample: Returned 0 rows, but expected 1 row

This is an error in dbt. Please try again. If the error persists, open an issue at https://github.com/dbt-labs/dbt-core
```

### Environment

```markdown
- OS: MacOS, Linux
- Python: 3.9
- dbt: 1.3.2
```

### Which database adapter are you using with dbt?

bigquery

**Comment by jtcohen6** [comment]
@nishantnathhoney I see you closed the issue - did you get to the bottom of this? Something else afoot?

**Comment by nishantnathhoney** [comment]
@jtcohen6 realized the issue was with specifying `materialized = 'table',` for bespoke tests which is not expected.
`meta` attribute wasn't the issue. Although improving the error messaging for `materialization` specified for bespoke tests would still be helpful - wasn't the crux for the issue I opened so I opted to close.

**Comment by jtcohen6** [comment]
@nishantnathhoney Got it! I believe we recently made exactly that UX improvement, and included it in v1.4: https://github.com/dbt-labs/dbt-core/issues/5229 / https://github.com/dbt-labs/dbt-core/pull/6025
